client/src/components/overview/image-gallery/DefaultView.jsx

```diff
@@ -34,66 +38,147 @@ class DefaultView extends React.Component {
+    }else{
+      this.setState({
+        imageIndex: selectedIndex,
+        currentPhoto: selectedPhoto,
+        image_id: image_id,
+        hiddenNext: false,
+        hiddenPrev: false
+      })
+    }
   }
 
-  cyclePhotos(dir) {
+  prevImage() {
     let currentStyleIndex = this.props.styleIndex;
     let photoIndex = this.state.imageIndex;
+    let prevPhoto = this.props.styles[currentStyleIndex].photos[photoIndex - 1]?.url;
+    let pPhoto = this.props.styles[currentStyleIndex].photos[photoIndex - 2]?.url;
 
-    if(dir.target.name === 'Next') {
-      let nextPhoto = this.props.styles[currentStyleIndex].photos[photoIndex + 1]?.url;
-      let image_id = nextPhoto.split('-')[1];
+    let image_id = prevPhoto?.split('-')[1];
 
-      if (nextPhoto !== undefined) {
-        this.setState({
-          currentPhoto: nextPhoto,
-          imageIndex: photoIndex + 1,
-          image_id: image_id
-        })
+      if (prevPhoto !== undefined) {
+        if(pPhoto === undefined) {
+          this.setState({
+            currentPhoto: prevPhoto,
+            imageIndex: photoIndex - 1,
+            image_id: image_id,
+            hiddenPrev: true
+          })
+        } else {
+          this.setState({
+            currentPhoto: prevPhoto,
+            imageIndex: photoIndex - 1,
+            image_id: image_id,
+            hiddenPrev: false,
+            hiddenNext: false
+          })
+        }
       }
-    } else {
-      let prevPhoto = this.props.styles[currentStyleIndex].photos[photoIndex - 1]?.url;
-      let image_id = prevPhoto.split('-')[1];
+  }
+  nextImage() {
+    let currentStyleIndex = this.props.styleIndex;
+    let photoIndex = this.state.imageIndex;
+    let nextPhoto = this.props.styles[currentStyleIndex].photos[photoIndex + 1]?.url;
+    let nPhoto = this.props.styles[currentStyleIndex].photos[photoIndex + 2]?.url;
 
-      if (prevPhoto !== undefined) {
-        this.setState({
-          currentPhoto: prevPhoto,
-          imageIndex: photoIndex - 1,
-          image_id: image_id
-        })
+    let image_id = nextPhoto?.split('-')[1];
+
+      if (nextPhoto !== undefined) {
+        if (nPhoto === undefined) {
+          this.setState({
+            currentPhoto: nextPhoto,
+            imageIndex: photoIndex + 1,
+            image_id: image_id,
+            hiddenNext: true
+          })
+        } else {
+          this.setState({
+            currentPhoto: nextPhoto,
+            imageIndex: photoIndex + 1,
+            image_id: image_id,
+            hiddenNext: false,
+            hiddenPrev: false
+          })
+        }
       }
+  }
+
+  cyclePhotos(e) {
+    e.preventDefault()
+    if(e.target.name === 'Next') {
+      this.nextImage()
+    } else {
+      this.prevImage()
     }
   }
 
   render() {
+    let nextButton;
+    let prevButton;
+    if (this.state.hiddenNext) {
+      nextButton = <></>
+    } else {
+      nextButton = (<button onClick={this.cyclePhotos} name="Next">➡️</button>)
```

Comment:
If you use CSS to set the button display to hidden, instead of using an empty fragment (when you don't want the button to show), the invisible button will still take up the same space on the screen. So when it toggles back and forth between shown/not shown, it won't move the rest of your items around.